File: Examples/CSharp/Loading-and-Saving/PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.cs

using System.IO;
using Aspose.Note;
using System;
using Aspose.Note.Saving;
namespace Aspose.Note.Examples.CSharp.Loading_Saving
{
    public class PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm
    {
        public static void Run()
        {
            
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadingAndSaving();

            // Load the document into Aspose.Note.
            Document doc = new Document(dataDir + "Aspose.one");
           
            var pdfSaveOptions = new PdfSaveOptions();

            // ExStart:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-1
            pdfSaveOptions.PageSplittingAlgorithm = new KeepPartAndCloneSolidObjectToNextPageAlgorithm(100);
            // ExEnd:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-1
            // Or
            // ExStart:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-2
            pdfSaveOptions.PageSplittingAlgorithm = new KeepPartAndCloneSolidObjectToNextPageAlgorithm(400);
            // ExEnd:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-2

            dataDir = dataDir + "PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm_out_.pdf";
            doc.Save(dataDir);
            Console.WriteLine("\nPage splitt successfully using keep part and clone solid object to next page algorithm.");
        }
    }
}

File: Examples/VisualBasic/Loading-and-Saving/PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.vb
Imports System.IO
Imports Aspose.Note
Imports Aspose.Note.Saving

Namespace Loading_Saving
    Public Class PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm
        Public Shared Sub Run()

            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadingAndSaving()

            ' Load the document into Aspose.Note.
            Dim doc As New Document(dataDir & Convert.ToString("Aspose.one"))

            Dim pdfSaveOptions = New PdfSaveOptions()

            ' ExStart:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-1
            pdfSaveOptions.PageSplittingAlgorithm = New KeepPartAndCloneSolidObjectToNextPageAlgorithm(100)
            ' ExEnd:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-1
            ' Or
            ' ExStart:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-2
            pdfSaveOptions.PageSplittingAlgorithm = New KeepPartAndCloneSolidObjectToNextPageAlgorithm(400)
            ' ExEnd:PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm.-2

            dataDir = dataDir & Convert.ToString("PageSplittUsingKeepPartAndCloneSolidObjectToNextPageAlgorithm_out.pdf")
            doc.Save(dataDir)
            Console.WriteLine(vbLf & "Page splitt successfully using keep part and clone solid object to next page algorithm.")
        End Sub
    End Class
End Namespace